File: src/java/org/jy/PeakIndexInMountainArray.java
package org.jy;

/**
 * 852. 山脉数组的峰顶索引
 * @author LJY
 */
public class PeakIndexInMountainArray {

    public int peakIndexInMountainArray(int[] A) {
        for (int i = 0; i < A.length - 1; i++) {
            if(A[i] > A[i + 1]){
                return i;
            }
        }
        return -1;
    }

}

File: src/scala/PeakIndexInMountainArray.scala
object org.jy.PeakIndexInMountainArray {

  def peakIndexInMountainArray(A: Array[Int]): Int = {
    for (i <- A.indices) {
      if (A(i) > A(i + 1)) return i
    }
    -1
  }

}
